lightningd/htlc_end.h

```diff
@@ -28,6 +28,14 @@ struct htlc_end {
 	u32 outgoing_cltv_value;
 	u32 cltv_expiry;
 	struct sha256 payment_hash;
+
+	/* If we are forwarding, remember the shared secret for an
+	 * eventual reply */
+	struct sha256 shared_secret;
```

Comment:
This could be a pointer, or we could actually just re-use the same "shared secret" pointer to hold this as the path_secrets.  But if so it'd need a comment as to the dual use!